chia/full_node/full_node.py

```diff
@@ -1079,11 +1081,15 @@ async def validate_block_batches(
                 peer, blocks = res
                 start_height = blocks[0].height
                 end_height = blocks[-1].height
-                success, state_change_summary = await self.add_block_batch(
+                success, state_change_summary, err = await self.add_block_batch(
                     blocks, peer, None if advanced_peak else uint32(fork_point_height), summaries
                 )
                 if success is False:
                     await peer.close(600)
+                    # check Chip 13 exception
+                    if err == Err.INVALID_POSPACE:
+                        self.add_to_bad_peak_cache(peak_hash, target_peak_sb_height)
+                        raise ValueError(f"block batch {start_height} to {end_height} failed Chip-13 validation")
```

Comment:
this is what we throw i dont really want to do in and change that, why 0013 ? is this how we refer to it in documentation ?

Reply:
Sorry I wasn't more clear.  I was suggesting that the error message was too prescriptive for the exception being checked for.  I would expect that many things that are not CHIP-0013 could result in `Err.INVALID_POSPACE` so a log message specifically stating that it _was_ a CHIP-0013 validation issue could be misleading in some cases.

Yes to the `CHIP-0013` form.

https://github.com/Chia-Network/chips/blob/609a750e35b958155ef3d559ca372f5be6a96395/CHIPs/chip-0001.md#chip-numbers
> When referring to a CHIP by number, it should be written in the hyphenated form CHIP-WXYZ where WXYZ is the CHIP's assigned number.

Reply:
i see, actually i was sure that in this path the only pospace failure possible is chip-0013 but i just took another look and you are right since for some reason we check chip-0013 in header validation and in body validation (i cant see a reason why this isn't redundant)

there is no way to add description to ERR (usually we wrap it in validation error for that), any ideas before i touch the return types ?

Reply:
i think the smallest change would be to add an Err type for CHIP-0013

Reply:
ok i think this looks better now